### PR TITLE
Marketing: Fix keyring user label and port to react-query

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

It seems that in #13335 we broke the keyring user label that appears when the connection is owned by another user. It has been broken for 4 years, never showing for anyone 😓 

This PR uses the opportunity to fix that bug.

Additionally, it removes one of the few remaining instances of the flux `UsersStore` and replaces it with a simple `react-query` instance that uses the existing `useUsersQuery` we already built.

This PR is part of #24150.

#### Testing instructions

* Go to `/marketing/connections/:site` for a site that has an active social network connection and you're the owner.
* Verify you still don't see the label, and everything works as it did before.
* Log in as another user that has admin permissions but isn't owning the connection.
* Verify you're seeing the label now and it shows the correct owner of the connection.
* Keep an eye on the console and verify we're not introducing any new errors.

Before:

After:

#### Notes

ESLint errors are expected, and are not being introduced in this PR.

### CODE REVIEW

**Comment by Tug** [comment]
Tried on calypso.live and it seems to be working fine 👍

**Comment by gwwar** [comment]
🤔 I'm not seeing the label on an Editor Test User:

Screen Shot 2021-04-08 at 2 19 54 PM

File: client/my-sites/marketing/connections/connection.jsx [hunk 1]
```diff
@@ -250,15 +239,37 @@ class SharingConnection extends Component {
...
 }
 
+const SharingConnectionKeyringUserLabel = localize(
+	( { siteId, keyringUserId, translate, userId } ) => {
+		const fetchOptions = {
+			search: keyringUserId,
+			search_columns: [ 'ID' ],
+		};
+
+		const { data } = useUsersQuery( siteId, fetchOptions );
```
Comment (gwwar) [inline client/my-sites/marketing/connections/connection.jsx:249]: This is undefined for the editor role and below:

Screen Shot 2021-04-08 at 2 31 36 PM

**Review by gwwar — APPROVED** [review]
Thanks for looking into this one @tyxla! I'm giving tentative approval. 

This works for me when viewing as another admin role:

Screen Shot 2021-04-08 at 2 33 34 PM

With the editor role and below a user doesn't have permissions to view the site users. 

Feel free to land after fixing that or land this first, then follow up in the next PR.

**Comment by tyxla** [comment]
Thanks for taking a look, @Tug and @gwwar!

> With the editor role and below a user doesn't have permissions to view the site users.

This seems to always have been the case - prior to this PR, we were still using the same endpoint, so editors were still unable to see the users. 

If you ask me, this limitation makes sense, too - according to WordPress's default roles and capabilities system, editors can't view users at all: https://wordpress.org/support/article/roles-and-capabilities/#editor

Therefore, I'm inclined to keep it as-is. But I'm happy to let you convince me otherwise 😉

**Pull request merged**